### ISSUE TITLE
Moves with sub rolls

### ISSUE THREAD

**Issue by asacolips** [asacolips-projects/dungeonworld#117]
In GitLab by @julien1205 on Mar 26, 2023, 07:36

Moves like backstab, where a 2nd roll is possible depending on the choosen option, dont work.
The second optional roll is made and revealed at the same time the initial roll is done while the intended behavior is to not know the 2nd roll  result until the right option is chosen.

**Comment by asacolips** [comment]
In GitLab by @geeklhem on Apr 30, 2023, 04:39

I suspect that the solution is just to make the subrolls "deferred inline rolls" instead of "immediate inline rolls" as [described in Foundry's documentation](https://foundryvtt.com/article/dice/).

In the case of the backstab, change the roll section in `src/packs/the-thief-moves/backstab.yml` to:

```
  choices: >-
    <ul>
    <li>You don’t get into melee with them</li>
    <li>You deal your damage + [[/roll 1d6]]</li>
    <li>You create an advantage, +1 forward to you or an ally acting on it</li>
    <li>Reduce their armor by 1 until they repair it</li>
    </ul>
```

by replacing the immediate roll `[[1d6]]`, with a deferred roll `[[/roll 1d6]]`. This creates an clickable button in the chat message to actually roll it: 

![image](https://gitlab.com/asacolips-projects/foundry-mods/dungeonworld/uploads/796acd5deb13c6425ef545e30f57a9fe/image.png)

After a quick look (`grep '\[\['`), all the rolls within moves are already differed rolls except for backstab, lay-on-hands, and acrane-arts. Maybe it's just a typo ?

In the case of Lay-on-hands it makes sense for the roll not to be deferred (it's not a choice), but for backstab and arcane-arts they should be deferred (so the player doesn't know the result of the roll before making the choice).

**Comment by asacolips** [comment]
In GitLab by @julien1205 on Apr 30, 2023, 17:21

Thanks to dear @geeklhem the issue is sovled on my fork in the dev branch.
Will MR at a later date.
The change is only applied to Dungeon World system compendium.
To solve this on an existing character the moves will have to be edited manually this way:
![manual_backstab_correct](https://gitlab.com/asacolips-projects/foundry-mods/dungeonworld/uploads/1ea87d5f7ae4f75df68cf7c45c3969d4/manual_backstab_correct.gif)

**Comment by asacolips** [comment]
In GitLab by @Asacolips on Jun 5, 2023, 16:42

Implemented in 1.7.0
